tests/rptest/tests/partition_balancer_test.py

```diff
@@ -159,7 +173,10 @@ def test_unavailable_nodes(self):
                 ns.step_and_wait_for_ready(node)
                 self.check_no_replicas_on_node(node)
 
-        self.run_validation()
+        # Restore the system to a fully healthy state before validation:
+        # not strictly necessary but simplifies debugging.
+        ns.make_available()
```

Comment:
Not necessary to do in this PR, but it would actually be great to make NodeStopper a context manager type (consumer validation is not the only place that can throw resulting in a confusing error message)